pkg/sql/plan2/build.go

```diff
@@ -35,6 +34,18 @@ func buildPlan(ctx CompilerContext, stmt tree.Statement) (*Query, error) {
 	return query, nil
 }
 
+func BuildPlan2(ctx CompilerContext, stmt tree.Statement) (*plan.Query, error) {
```

Comment:
I think you can move /explain/buildexplain_test.go to  /plan2/    

then there will be not necessary to  add BuildPlan2() and NewMockOptimizer2()